controllers/services/monitoring/monitoring_support.go

```diff
@@ -0,0 +1,70 @@
+package monitoring
+
+import (
+	"context"
+	"fmt"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"k8s.io/utils/set"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	cr "github.com/opendatahub-io/opendatahub-operator/v2/pkg/componentsregistry"
+	odhcli "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/client"
+)
+
+const (
+	prometheusConfigurationEntry = "prometheus.yml"
+)
+
+type PrometheusConfig struct {
+	RuleFiles []string       `yaml:"rule_files"`
+	Others    map[string]any `yaml:",inline"`
+}
+
+func (pc *PrometheusConfig) computeRules(
+	ctx context.Context,
+	cli *odhcli.Client,
+	registry *cr.Registry,
+) error {
+	// Map component names to their rule prefixes
+	dsc, err := cluster.GetDSC(ctx, cli)
+	if err != nil {
+		return fmt.Errorf("failed to get DataScienceCluster instance: %w", err)
+	}
+
+	s := set.New(pc.RuleFiles...)
+
+	err = registry.ForEach(func(ch cr.ComponentHandler) error {
+		rule, ok := componentRules[ch.GetName()]
+		if !ok {
+			return nil
+		}
+
+		ms := ch.GetManagementState(dsc)
+		switch ms {
+		case operatorv1.Removed:
+			s = s.Delete(rule)
+		case operatorv1.Managed:
+			ci := ch.NewCRObject(dsc)
+			ready, err := cluster.IsReady(ctx, cli, ci)
```

Comment:
not about this PR, but is the old logic correct:
if component X is enabled in DSC, but it never gets ready (crashloop) then promrule will never be added. so SRE will never get alerts?

Reply:
> to avoid having misfired alerts when the components is being enabled for the first time

And I still have the concern about unconditional deploying of scraping jobs even without the rules.

Reply:
everything is bad, thats why we need to get this https://github.com/opendatahub-io/opendatahub-operator/pull/1676 out, one less headache in long term :laughing:

Reply:
> hover we do lack other mechanisms and alerts to cover the case you mentioned.

@lburgazzoli do we have an issue opened that capture it?